mlflow/sklearn/utils.py

```diff
@@ -1,7 +1,11 @@
-from distutils.version import LooseVersion
 import inspect
-from itertools import islice
 import logging
+import mlflow
+import sklearn
```

Comment:
FYI: We have the check (added by #3288) to assert that `import mlflow` works with its default depedencies: https://github.com/mlflow/mlflow/blob/70b61e309a2a4755d616ea3dac4f506917e98439/.github/workflows/master.yml#L187-L199

Currently, this check doesn't fail because `import ... from mlflow.sklearn.utils` is executed in the `autolog` function, but as @dbczumar suggested, we should import these optional dependencies in functions.

Reply:
This is really a good catch! Thanks for the suggestion!